ingenialink/dictionary.py

```diff
@@ -1887,3 +1902,7 @@ def _append_missing_registers(
         if self.__MON_DIST_STATUS_REGISTER in self._registers[0]:
             for register in self._monitoring_disturbance_registers:
                 self._add_register_list(register)
+
+            if self.interface in [Interface.CAN, Interface.ECAT]:
+                for obj in self._monitoring_disturbance_objects:
+                    self._add_canopen_object(obj)
```

Comment:
why don't we place add canopen object to canopendictionary?
ethercat inherits from it.
And override append missing registers instead of doing check with interface